share/spice/geometry/geometry.js

```diff
@@ -448,7 +451,13 @@
                 break;
             }
         }
-
+        //fails if the search string asks for a formula that does not belong to the shape,
+        //unless the query is triggered by the generic keywords 'geometry' and 'formulas'
+        if(found_formula == false && !(query.match("geometry") || query.match("formulas"))){
```

Comment:
This doesn't feel like it will scale well with more triggers - perhaps we should move it out to an Array that we can then check against.